### ISSUE TITLE
Improve uncaught exception handler double logging

### ISSUE THREAD

**Issue by cowwoc** [eclipse/jetty.project#3317]
HttpChannel line 583 reads:

```
// No stack trace unless there is debug turned on
LOG.warn("{} {}",_request.getRequestURI(), no_stack.toString()); 
if (LOG.isDebugEnabled())
  LOG.debug(_request.getRequestURI(), failure);
```

If `WARN` is enabled, we get a truncated warning (no stacktrace) but if `DEBUG` is on we get the message twice: once with a stacktrace and one without. Worse yet, if we enable `DEBUG` we get many other debug-level logs from `HttpChannel` that we might not have interest in.

I am expecting:

1. If uncaught exceptions are logged, it is always with the `WARN` level.
2. Users should be able to enable/disable logging of uncaught exceptions independent of other warnings (meaning, either use a separate logger for this case or add a flag for us to enable/disable logging).
3. We should have a flag for logging stacktraces or not for uncaught exceptions (independent of the logging level).

**Comment by gregw** [comment]
This logic is for exceptions: `BadMessageException`, `IOException`, `TimeoutException`, which can all occur on the server due to bad clients. At best this allows bad clients to create a lot of noise in the logs. At worst it can malicious clients to have a slight DOS vector as logging stack traces can consume CPU,  file space and other resources (including developers on the jetty team handling false positive reports of "jetty exceptions").

These exceptions are not uncaught, as they are caught by jetty as it handles requests received by clients and passed to servlets etc.  So Jetty is handling the exceptions by making an assessment of log severity and verbosity.

What if we altered the code to:
```
if (LOG.isDebugEnabled())
  LOG.debug(_request.getRequestURI(), failure);
else
  LOG.warn("{} {}",_request.getRequestURI(), no_stack.toString()); 
```

**Comment by cowwoc** [comment]
Hi @gregw,

The proposed change looks good in that it meets point 1 and 3 on my list. I would still like it to use a separate logger (point 2).

We can go a step further by allowing users to plug in their own exception handler. Wrap the proposed implementation in a default handler, and let users provide their implementation if they so wish.

**Comment by joakime** [comment]
Fixed in commit 0809228549c8dbb7ffede1ccd6d4ff57d6a60474
